tailfs/tailfsimpl/dirfs/dirfs_test.go

```diff
@@ -0,0 +1,353 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package dirfs
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/tailscale/xnet/webdav"
+	"tailscale.com/tailfs/tailfsimpl/shared"
+	"tailscale.com/tstest"
+)
+
+func TestStat(t *testing.T) {
+	cfs, _, _, clock, close := createFileSystem(t)
+	defer close()
+
+	tests := []struct {
+		label    string
+		name     string
+		expected fs.FileInfo
+		err      error
+	}{
+		{
+			label: "root folder",
+			name:  "",
+			expected: &shared.StaticFileInfo{
```

Comment:
Since the fields of the `StaticFileInfo` are all exported, we might get nicer results using `github.com/google/go-cmp/cmp` to print the test results:

```go
if diff := cmp.Diff(fi, test.want); diff != "" {
   t.Errorf("Wrong file info (-got, +want):\n%s", diff)
}
```

that already handles printing a nice diff in case where parts of the structure are wrong.